src/Raven.Server/Utils/ReplicationUtils.cs

```diff
@@ -127,6 +132,11 @@ public static class ReplicationUtils
             {
                 var getTcpInfoCommand = new GetTcpInfoCommand();
                 requestExecuter.Execute(getTcpInfoCommand, context);
+
+                //notfound status also sets command's Result to null, so this prevents NRE
+                if (getTcpInfoCommand.StatusCode == HttpStatusCode.NotFound)
```

Comment:
No, this is wrong. 
You are handling a single failure scenario, and ignoring all else, which is very wrong.
What happens if this is a 403 response? What happens if the endpoint sent garbage back?